sphinx/util/i18n.py

```diff
@@ -207,6 +207,7 @@ def format_date(
             # Check if we have to use a different babel formatter then
             # format_datetime, because we only want to format a date
             # or a time.
+            function: Callable[..., str]
```

Comment:
It depends on whether we use `format_date`, `format_time` or `format_datetime`. The `babel_format_date` function itself assumes just a `Callable` because it makes some hacky injection (which I prefer not to change in this PR).

Reply:
I'll detail this.